ofrak_core/ofrak/resource.py

```diff
@@ -1527,6 +1530,21 @@ async def summarize_tree(
             tree_string += f"{indent}{branch_symbol}{SPACER_LINE}{child_tree_string}"
         return tree_string
 
+    @asynccontextmanager
+    async def temp_to_disk(
+        self,
+        prefix: Optional[str] = None,
+        suffix: Optional[str] = None,
+        dir: Optional[str] = None,
+        delete: bool = True,
+    ) -> AsyncIterator[str]:
+        with tempfile.NamedTemporaryFile(
+            mode="wb", prefix=prefix, suffix=suffix, dir=dir, delete_on_close=False, delete=delete
+        ) as temp:
```

Comment:
Do you think it makes sense to let additional keyword arguments fall through here? I lean towards passing along all `kwargs`, but don't feel strongly either way.

Reply:
the only other argument we _might_ to pass is `dir`. We don't want to pass stuff like `encoding` through, since it specifically is creating a copy of the resource bytes on disk.

Reply:
wait I'm already passing `dir`. yeah in that case we shouldn't pass through anything else